packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -113,108 +114,111 @@ const withInfiniteScroll = (
         showSeeMore: false,
       };
 
-      this.onItemsFetch = props.onItemsFetch;
-      this.observer = new IntersectionObserver(this.handleIntersection, {
-        threshold: 1,
-      });
-    }
-
-    componentDidMount() {
-      if (this.sentinel) {
-        this.observer.observe(this.sentinel);
-      }
+      this.props.dataSource.onDataChange(this.updateData);
```

Comment:
this needs to be done again in `componentWillReceiveProps` if the DataSource changes, and we need to call `offDataChange` in the old data source 

We also need to call `offDataChange` on `componentWillUnmount`

Reply:
isn't `componentWillReceiveProps` going to be deprecated?

Reply:
it will - we should probably use `componentDidUpdate` then, and also reset the whole scroll state if that happens as changing the datasource possibly means completely different data.